lib/src/main/java/net/ypresto/androidtranscoder/MediaTranscoder.java

```diff
@@ -176,6 +185,9 @@ public void run() {
                     Log.w(TAG, "Transcode failed: input file (fd: " + inFileDescriptor.toString() + ") not found"
                             + " or could not open output file ('" + outPath + "') .", e);
                     caughtException = e;
+                } catch (InterruptedException e) {
+                    Log.i(TAG, "Cancel transcode video file.", e);
+                    caughtException = e;
                 } catch (RuntimeException e) {
                     Log.e(TAG, "Fatal error while transcoding, this might be invalid format or bug in engine or Android.", e);
                     caughtException = e;
```

Comment:
Please rethrow `e` after handler.post()` to notify the Future.
You should use `Callable` instead of `Runnable` to do so.